cip-TOKEN-STD/cip-meiersi-Canton-Network-Token-Standard.md

```diff
@@ -0,0 +1,567 @@
+be signed by their validator node. Registries should aim to allow for up to 24 hours
+between the preparation of such a transaction and their submission to the network.
+
+##### Registry Specific UIs
+
+The standard allows registries to communicate the URL for a registry-specific
+UI to wallet clients. Redirecting the user to this URL serves as an escape hatch
+for serving registry specific data and driving registry-specific workflows.
+
+The URL for the registry-specific UI could for example be used for the user
+to ask for the redemption of a wrapped token to a specific address on the
+original chain/ledger. Providing this URL allows universal wallet clients
+to handle all steps of common workflows, while allowing registries to
+implement custom workflows where required.
+
+Furthermore, registries that do not use Canton as the authoritative ledger
+might decide to not represent holdings on Canton. This is still valuable,
+as these assets can be used in DVP workflows like any other asset.
+The wallet clients would show real-time information about all in-progress transfers.
+Instead of showing the holdings they would though show a redirect to the registries
+own UI for querying the holding balance.
+
+The standard does not define an interface to query the holdings of such registries
+from a Canton wallet client, as that would require standardizing how to authenticate
+such a call.
+
+
+#### Canton Coin Implementation
+
+Canton Coin (CC) implements all APIs of the standard.
+FOP transfers of CC require receivers to have preapproved incoming transfers from any sender.
+The are completed as part of the single Daml transaction instructing them.
+Likewise, DVP allocations of CC are created within a single Daml transaction instructing them.
+No preapproval by the receiver is required, as that authorization is expected to
+be funneled through the apps' settlement worklows.
+
+CC is standards compliant except for the following two limitations:
+
+- *short submission delay*: the CC implementation only supports a 1 minute
+  delay between preparing a transaction and submitting it to the network.
+  This is in contrast to the 24h delay targeted by the standard.
+- *no holding fees in portfolio view*: the amounts of CC holdings always show
+  the amount as of the round that they were created in. Wallets cannot
+  use the standardized APIs to display holding fees for CC holdings, but they
+  can read the `Amulet` contracts themselves if they desire to show holding fees.
+
+Both of these limitations are expected to be removed in the future, as
+explained in [Canton Coin Limitations](#canton-coin-limitations).
+
+
+### Details
+
+The standard defines six APIs. Each consists of an on-ledger Daml API and
+optionally an off-ledger HTTP API. The six APIs and their purpose are:
+
+- *token metadata API*: serve symbol name, total supply, URL for registry-specific UI, other metadata
+- *holdings API*: populate portfolio view
```

Comment:
TODO: check if burn/mint fits in here. Probably does when you view it from the point of enabling wallets to display a decent history.